### PR TITLE
slint-lsp: 1.5.1 -> 1.8.0

### PR DESCRIPTION
**Pull request by xgroleau**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Tested the build and briefly the LSP

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by PaulDance — APPROVED** [review]
Looks good to me!

(although I have absolutely no say in this whatsoever :hand_over_mouth:)

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351029`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>slint-lsp</li>
  </ul>
</details>

File: pkgs/development/tools/misc/slint-lsp/default.nix
Comment (Scrumplex) [inline pkgs/development/tools/misc/slint-lsp/default.nix:1]: I'd recommend moving this file to `by-name` so you can use the merge bot in the future.

File: pkgs/development/tools/misc/slint-lsp/default.nix [hunk 1]
```diff
@@ -46,6 +46,8 @@ rustPlatform.buildRustPackage rec {
     OpenGL
   ];
 
+  doCheck = false;
```
Comment (Scrumplex) [inline pkgs/development/tools/misc/slint-lsp/default.nix:49]:
```suggestion
  # rustc throws the following error:
  # use of undeclared crate or module `i_slint_backend_testing`
  doCheck = false;
```

**Review by Scrumplex — COMMENTED** [review]
Builds and runs. Tested by formatting a .slint file.

File: pkgs/development/tools/misc/slint-lsp/default.nix
Comment (xgroleau) [reply pkgs/development/tools/misc/slint-lsp/default.nix:1]: Thanks for the headsup, i've moved it to by-name directory

**Comment by xgroleau** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by xgroleau** [comment]
Ahh it's only for pacakge edit, which makes total sense. Fair enough, that will still allow for future PR's

File: pkgs/by-name/sl/slint-lsp/package.nix [hunk 2]
```diff
@@ -46,10 +46,13 @@ rustPlatform.buildRustPackage rec {
...
   ];
 
+  doCheck = false;
+
   postInstall = lib.optionalString stdenv.hostPlatform.isLinux  ''
     patchelf --set-rpath ${lib.makeLibraryPath rpathLibs} $out/bin/slint-lsp
   '';
 
+  # Tests requires `i_slint_backend_testing` which is only a dev dependency
   dontPatchELF = true;
```
Comment (Scrumplex) [inline pkgs/by-name/sl/slint-lsp/package.nix:56]: You inserted the comment for the wrong attribute :sweat_smile:

Comment (xgroleau) [reply pkgs/by-name/sl/slint-lsp/package.nix]: woops, rough morning today!

**Review by Scrumplex — APPROVED** [review]

**Comment by PaulDance** [comment]
Thanks!

**Pull request merged**